plugins/renepay/pay.c

```diff
@@ -1138,14 +1140,10 @@ static struct command_result *json_pay(struct command *cmd,
 					       *riskfactor_millionths,
 					       *min_prob_success_millionths,
 					       use_shadow);
-	tal_add_destructor2(renepay,
-			    renepay_cleanup,
-			    pay_plugin->gossmap);
 
-	/* We inmediately add this payment to the payment list. */
-	tal_steal(pay_plugin->ctx, renepay->payment);
-	list_add_tail(&pay_plugin->payments, &renepay->payment->list);
-	tal_add_destructor(renepay->payment, destroy_payment);
+	/* We immediately add this payment to the payment list. */
+	list_add_tail(&pay_plugin->payments, &payment->list);
```

Comment:
I agree that having a single structure to represent a payment is the way to go. But we need to keep in mind for a future optimization to keep a list of payment information that is relevant to `renepaystatus` and nothing else. Otherwise we keep extra data that is no longer used.

Reply:
Definately! paystatus is fairly useless as is, though, and we need to think hard about what should actually go there. It could just be the paynotes, and the final status really.